packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx

```diff
@@ -114,81 +132,39 @@ class AudioPlayerWithYoutubeSpotify extends Component {
-    const newSourceAlbumInfo = albumSpotifyYoutubeInfo[newSource];
-    const noAlbumWithNewSource = currentTrack === 0 && !newSourceAlbumInfo;
-    const noTrackWithNewSource = !tracklistToShow.find(f => currentTrack === f.trackNumber);
-
-    const firstTrackAvailable = tracklistToShow.find(f => f.trackNumber === 1) ? tracklistToShow.find(f => f.trackNumber === 1) : head(tracklistToShow);
-    const { trackNumber: availableTrackNumber } = firstTrackAvailable;
-    const trackSelected = (noAlbumWithNewSource || noTrackWithNewSource) ? availableTrackNumber : currentTrack;
-
-    const newState = { channelToPlay, tracklistToShow, trackSelected };
-    this.setState(newState);
-  }
-
-  /**
-   * Callback every time JWPlayer plays a track
-   * @param { object } playlistItem - JWPlayer track object
-   */
-  jwplayerPlaylistChange(playlistItem) {
-    const { newTrackIndex } = playlistItem;
-    this.setState({ trackSelected: newTrackIndex + 1 });
-  }
-
-  jwplayerStartingPoint(index) {
-    this.setState({ trackStartingPoint: index + 1 });
-  }
-
-  /**
-   * Callback every time user selects a track from the tracklist
-   * @param { object } event - React synthetic event
-   */
-  selectThisTrack(event) {
-    const selected = event.currentTarget;
-    const selectedTrackNumber = parseInt(selected.getAttribute('data-track-number'), 10);
-
-    this.setState({
-      trackSelected: Number.isInteger(selectedTrackNumber)
-        ? selectedTrackNumber
-        : 1
-    });
+    this.setState({ channelToPlay, tracklistToShow }, this.updateURL);
   }
 
   /**
    * Find the right audio metadata depending on the channel and track number
    */
   getAudioSourceInfoToPlay() {
     const {
-      albumData, channelToPlay, trackSelected, tracklistToShow
+      albumData, channelToPlay, trackSelected, tracklistToShow, trackStartingPoint
     } = this.state;
-    if (trackSelected === null) return {};
-
     const isAlbum = trackSelected === 0;
-    let audioSource = null;
-
-    if (channelToPlay !== 'archive') {
-      // it's youtube or spotify
-      const { albumSpotifyYoutubeInfo } = albumData;
-      const albumSource = albumSpotifyYoutubeInfo[channelToPlay] || null;
-      if (isAlbum) {
-        audioSource = {};
-        audioSource[channelToPlay] = albumSource;
-      }
 
-      if (!audioSource) {
-        audioSource = find(tracklistToShow, track => track.trackNumber === trackSelected);
-      }
+    let audioSource;
 
-      return audioSource || {};
+    if (channelToPlay === 'archive') {
+      // ia jw player only needs index
+      audioSource = {
+        index: trackSelected
+      };
+      return audioSource;
     }
 
-    // ia jw player only needs index
-    audioSource = {
-      index: trackSelected
-    };
+    // it's youtube or spotify
+    const { albumSpotifyYoutubeInfo } = albumData;
+    const albumSource = albumSpotifyYoutubeInfo[channelToPlay] || null;
+    if (isAlbum) {
+      audioSource = {};
```

Comment:
**Suggestion**: merge these two statements into one with the new property syntax available.

```javascript
audioSource = { [channelToPlay]: albumSource }
```